checkers/Board.py

```diff
@@ -195,7 +245,161 @@ def find_valid_moves_recursive(self, player_turn):
                                 self.piece_set.add(
                                     (new_right_grid_row, new_right_grid_col))
 
-            return self.find_valid_moves_recursive(player_turn)
+            return self.find_man_valid_moves(player_turn)
+
+    def find_king_valid_moves(self, player_turn):
```

Comment:
Not sure about this function yet, if it's possible, combing both man and king functions might be nice but need to find a way to shorten the code. Need to discuss about this.